### ISSUE TITLE
Failed to re-do node discovery after changing node IP address

### ISSUE THREAD

**Issue by robin2008** [xcat2/xcat-core#4890]
Customer report below issue:
1, To follow xcat doc define nodes and do discovery, it works well.
2, Find the IP address is configured a wrong value, then fix the IP address and try a new node discovery
    - chdef xxx ip=<new>
    - makehosts xxx
    - makedns -n xxx
3, After node discovered in genesis,  finds that the dhcp lease cannot be set to new IP address

**Comment by robin2008** [comment]
After deep investigation, it is caused by a cache in NetworkUtils.pm (`%::hostiphash`)
The cache is designed to improve the process-level name  resolution, and it works fine in makedhcp/makedns, etc.  

But unfortunately,  for discovery,  it's plugins is run in the same process, which means the cache will not stale, and keep the old value even if the value is changed.

Reproduce steps:
1, define a node with an IP(Assume 192.168.139.16), makehosts/makedns and then  discover it.  (This make sure the cache has the name and IP)
2, change the IP address of this node, and discover it again.
```
# chdef mid05tor12cn16 ip=192.168.149.16
1 object definitions have been created or modified.
# makehosts mid05tor12cn16
# makedns -n mid05tor12cn16
Handling mid05tor12cn16 in /etc/hosts.
Getting reverse zones, this may take several minutes for a large cluster.
Completed getting reverse zones.
Updating zones.
Completed updating zones.
Restarting named
Restarting named complete
Updating DNS records, this may take several minutes for a large cluster.
Completed updating DNS records.
DNS setup is completed
# nslookup mid05tor12cn16
Name:   mid05tor12cn16.example.com
Address: 192.168.149.16
```
3, After discovered `findme` request is handled, check the new IP in genesis or in the dhcp lease file
The IP is still 192.168.139.16,
```
makedhcp -q mid05tor12cn16
mid05tor12cn16: ip-address = 192.168.139.16, hardware-address = 11:22:33:44:55:66
```

**Comment by robin2008** [comment]
It is decided that clear the cache when handling new incoming `findme`.
The workaround is addressed by PR https://github.com/xcat2/xcat-core/pull/4891

**Comment by robin2008** [comment]
"install monitor" process may have the similar issue, and it is more complex, plan to fix it in 2.14
